packages/webpack/template-webpack-plugin/src/LynxTemplatePlugin.ts

```diff
@@ -681,23 +681,29 @@ class LynxTemplatePluginImpl {
       compilation,
     )!;
 
-    await Promise.all(
-      Object.entries(asyncChunkGroups).map(([entryName, chunkGroups]) => {
-        const chunkNames =
-          // We use the chunk name(provided by `webpackChunkName`) as filename
-          chunkGroups
-            .map(cg => hooks.asyncChunkName.call(cg.name))
-            .filter(chunkName => chunkName !== undefined);
-
-        const filename = Array.from(new Set(chunkNames)).join('_');
-
-        // If no filename is found, avoid generating async template
-        if (!filename) {
-          return;
-        }
+    // merge chunk groups with the same output `filename`
+    const filename2ChunkGroups: Record<string, ChunkGroup[]> = {};
```

Comment:
By the screenshot you provided in the issue, it seems like an issue of the `getAsyncChunkGroups` method that we did not successfully `groupBy` the two background and main-thread `chunkGroup`.

So I think it would be reasonable to fix the problem there. Maybe you can group by `cg.name` instead of `origin.imports`.